File: src/engine/commands/closecommand.cpp
#include "closecommand.h"


#define super Command

CloseCommand::CloseCommand(QObject *parent) :
    super(parent) {

    setDescription("Close an exit, typically a door.\n"
                   "\n"
                   "Example: close door");
}

CloseCommand::~CloseCommand() {
}

void CloseCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtr exit = takeObject(currentRoom()->exits());
    if (!requireSome(exit, "Close what?")) {
        return;
    }

    player->close(exit);
}

File: data/commands/closecommand.js
function CloseCommand() {
    Command.call(this);

    this.setDescription("Close an exit, typically a door or a window.\n" +
                        "\n" +
                        "Example: close door");
}

CloseCommand.prototype = new Command();
CloseCommand.prototype.constructor = CloseCommand;

CloseCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var portal = this.takeObject(this.currentRoom.portals);
    if (!this.requireSome(portal, "Close what?")) {
        return;
    }

    player.close(portal);
};

CommandRegistry.registerCommand("close", new CloseCommand());
